database/mysql/mysql.go

```diff
@@ -98,43 +97,37 @@ func WithInstance(instance *sql.DB, config *Config) (database.Driver, error) {
 	return mx, nil
 }
 
-// urlToMySQLConfig takes a net/url URL and returns a go-sql-driver/mysql Config.
-// Manually sets username and password to avoid net/url from url-encoding the reserved URL characters
-func urlToMySQLConfig(u nurl.URL) (*mysql.Config, error) {
-	origUserInfo := u.User
-	u.User = nil
-
-	c, err := mysql.ParseDSN(strings.TrimPrefix(u.String(), "mysql://"))
+func (m *Mysql) Open(url string) (database.Driver, error) {
+	config, err := mysql.ParseDSN(strings.TrimPrefix(url, "mysql://"))
 	if err != nil {
 		return nil, err
 	}
-	if origUserInfo != nil {
-		c.User = origUserInfo.Username()
-		if p, ok := origUserInfo.Password(); ok {
-			c.Passwd = p
-		}
-	}
-	return c, nil
-}
 
-func (m *Mysql) Open(url string) (database.Driver, error) {
-	purl, err := nurl.Parse(url)
+	config.MultiStatements = true
+
+	// Keep backwards compatibility from when we used net/url.Parse() to parse the DSN.
+	// net/url.Parese() would automatically unescape it for us.
+	// See: https://play.golang.org/p/q9j1io-YICQ
+	user, err := nurl.QueryUnescape(config.User)
 	if err != nil {
 		return nil, err
 	}
+	config.User = user
 
-	q := purl.Query()
-	q.Set("multiStatements", "true")
-	purl.RawQuery = q.Encode()
+	password, err := nurl.QueryUnescape(config.Passwd)
+	if err != nil {
+		return nil, err
+	}
+	config.Passwd = password
 
-	migrationsTable := purl.Query().Get("x-migrations-table")
+	migrationsTable := config.Params["x-migrations-table"]
```

Comment:
Using `config.Params` works for now, but because we're no longer pre-processing the DSN, if for some reason a `x-*` parameter collides with a go-sql-driver/mysql param, then we'll lose it. This is probably fine for now, but something to be wary of.